### ISSUE TITLE
Upgrade from older version failing

### ISSUE THREAD

**Issue by Nedlinin** [oerdnj/deb.sury.org#315]
nedlinin@nedlinin:~$ sudo apt-get upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
3 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n]
Setting up php7.0-cli (7.0.5-1+deb.sury.org~trusty+1) ...
Error: The new file /usr/lib/php/7.0/php.ini-production.cli does not exist!
dpkg: error processing package php7.0-cli (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of php7.0-dev:
 php7.0-dev depends on php7.0-cli (>= 7.0.5-1+deb.sury.org~trusty+1); however:
  Package php7.0-cli is not configured yet.

dpkg: error processing package php7.0-dev (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                                                                                                          dpkg: dependency problems prevent configuration of php7.0-fpm:
 php7.0-fpm depends on php7.0-cli; however:
  Package php7.0-cli is not configured yet.

dpkg: error processing package php7.0-fpm (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                                                                                                          Errors were encountered while processing:
 php7.0-cli
 php7.0-dev
 php7.0-fpm
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by Sopsys** [comment]
Confirmed, same issue when performing a clean install after purge.

root@server:~# apt-get install php7.0-cli
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  php7.0-common php7.0-json php7.0-opcache php7.0-readline
Suggested packages:
  php-pear
The following NEW packages will be installed:
  php7.0-cli php7.0-common php7.0-json php7.0-opcache php7.0-readline
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 1,840 kB of archives.
After this operation, 9,045 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ppa.launchpad.net/ondrej/php/ubuntu/ trusty/main php7.0-common amd64 7.0.5-1+deb.sury.org~trusty+1 [487 kB]
Get:2 http://ppa.launchpad.net/ondrej/php/ubuntu/ trusty/main php7.0-json amd64 7.0.5-1+deb.sury.org~trusty+1 [17.0 kB]
Get:3 http://ppa.launchpad.net/ondrej/php/ubuntu/ trusty/main php7.0-opcache amd64 7.0.5-1+deb.sury.org~trusty+1 [76.0 kB]
Get:4 http://ppa.launchpad.net/ondrej/php/ubuntu/ trusty/main php7.0-readline amd64 7.0.5-1+deb.sury.org~trusty+1 [12.2 kB]
Get:5 http://ppa.launchpad.net/ondrej/php/ubuntu/ trusty/main php7.0-cli amd64 7.0.5-1+deb.sury.org~trusty+1 [1,248 kB]
Fetched 1,840 kB in 0s (8,694 kB/s)
Selecting previously unselected package php7.0-common.
(Reading database ... 86645 files and directories currently installed.)
Preparing to unpack .../php7.0-common_7.0.5-1+deb.sury.org~trusty+1_amd64.deb ...
Unpacking php7.0-common (7.0.5-1+deb.sury.org~trusty+1) ...
Selecting previously unselected package php7.0-json.
Preparing to unpack .../php7.0-json_7.0.5-1+deb.sury.org~trusty+1_amd64.deb ...
Unpacking php7.0-json (7.0.5-1+deb.sury.org~trusty+1) ...
Selecting previously unselected package php7.0-opcache.
Preparing to unpack .../php7.0-opcache_7.0.5-1+deb.sury.org~trusty+1_amd64.deb ...
Unpacking php7.0-opcache (7.0.5-1+deb.sury.org~trusty+1) ...
Selecting previously unselected package php7.0-readline.
Preparing to unpack .../php7.0-readline_7.0.5-1+deb.sury.org~trusty+1_amd64.deb ...
Unpacking php7.0-readline (7.0.5-1+deb.sury.org~trusty+1) ...
Selecting previously unselected package php7.0-cli.
Preparing to unpack .../php7.0-cli_7.0.5-1+deb.sury.org~trusty+1_amd64.deb ...
Unpacking php7.0-cli (7.0.5-1+deb.sury.org~trusty+1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up php7.0-common (7.0.5-1+deb.sury.org~trusty+1) ...

Creating config file /etc/php/7.0/mods-available/calendar.ini with new version

Creating config file /etc/php/7.0/mods-available/ctype.ini with new version

Creating config file /etc/php/7.0/mods-available/exif.ini with new version

Creating config file /etc/php/7.0/mods-available/fileinfo.ini with new version

Creating config file /etc/php/7.0/mods-available/ftp.ini with new version

Creating config file /etc/php/7.0/mods-available/gettext.ini with new version

Creating config file /etc/php/7.0/mods-available/iconv.ini with new version

Creating config file /etc/php/7.0/mods-available/pdo.ini with new version

Creating config file /etc/php/7.0/mods-available/phar.ini with new version

Creating config file /etc/php/7.0/mods-available/posix.ini with new version

Creating config file /etc/php/7.0/mods-available/shmop.ini with new version

Creating config file /etc/php/7.0/mods-available/sockets.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvmsg.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvsem.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvshm.ini with new version

Creating config file /etc/php/7.0/mods-available/tokenizer.ini with new version
Setting up php7.0-json (7.0.5-1+deb.sury.org~trusty+1) ...

Creating config file /etc/php/7.0/mods-available/json.ini with new version
Setting up php7.0-opcache (7.0.5-1+deb.sury.org~trusty+1) ...

Creating config file /etc/php/7.0/mods-available/opcache.ini with new version
Setting up php7.0-readline (7.0.5-1+deb.sury.org~trusty+1) ...

Creating config file /etc/php/7.0/mods-available/readline.ini with new version
Setting up php7.0-cli (7.0.5-1+deb.sury.org~trusty+1) ...
update-alternatives: using /usr/bin/php7.0 to provide /usr/bin/php (php) in auto mode
update-alternatives: using /usr/bin/phar7.0 to provide /usr/bin/phar (phar) in auto mode
update-alternatives: using /usr/bin/phar.phar7.0 to provide /usr/bin/phar.phar (phar.phar) in auto mode
Error: The new file /usr/lib/php/7.0/php.ini-production.cli does not exist!
dpkg: error processing package php7.0-cli (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 php7.0-cli
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by oerdnj** [comment]
Right, it will be (hopefully) fixed in the next upload I just did.

**Comment by rydhoms** [comment]
Yeah hope fixed soon.. migrating to older version now

**Comment by oerdnj** [comment]
It's building right now - https://launchpad.net/~ondrej/+archive/ubuntu/php/+build/9423675

That was a unfortunate side-effect of preparing package for source-only uploads to Debian. Sorry for the disturbance.

**Comment by browner12** [comment]
thanks for the quick turnaround looking into this.  will it be as simple as running `apt-get update` `apt-get upgrade` again? or will we have to do anything else?

**Comment by Senn0** [comment]
I would like to know this as well :-)

**Comment by Nedlinin** [comment]
@browner12 @Senn0 : 

New build works fine on my end.  To force an update now:

    sudo rm -rf /var/lib/apt/lists/*
    sudo apt-get update

Then try installing it/upgrading it in your typical fashion.

These commands remove the old cached copy of the software from your machine (first command) and the second one simply asks to retrieve them again.  Please note this is a bit heavy handed to remove the old lists but there should not be any side effects except for the sudo apt-get update process taking a bit longer than normal (as it has to download *all* lists again)

**Comment by oerdnj** [comment]
@browner12 @Senn0 Yep, just `apt-get update && apt-get upgrade`, fixed now:
```
Setting up php7.0-cli (7.0.5-2+deb.sury.org~trusty+1) ...
update-alternatives: using /usr/bin/php7.0 to provide /usr/bin/php (php) in auto mode
update-alternatives: using /usr/bin/phar7.0 to provide /usr/bin/phar (phar) in auto mode
update-alternatives: using /usr/bin/phar.phar7.0 to provide /usr/bin/phar.phar (phar.phar) in auto mode

Creating config file /etc/php/7.0/cli/php.ini with new version
```

**Comment by oerdnj** [comment]
@Nedlinin I don't think there's a need to remove the old lists at all. It should happen automatically.

**Comment by Nedlinin** [comment]
@oerdnj : It is supposed to but for some reason my weren't coming in updated.  I cleared old lists just to check and it started working.  Highly recommend the standard method first though!

**Comment by rydhoms** [comment]
Worked now. Thanks @oerdnj

**Comment by oerdnj** [comment]
@Nedlinin Strange, anyway - it's harmless to purge just the `rm -f /var/lib/apt/lists/*php*` and then do `apt-get update` again.

**Comment by franchb** [comment]
Thank you, @oerdnj ! Now it all works!

**Comment by johnluxor** [comment]
Hello.
This problem for me

Setting up php7.0-fpm (7.0.6-3+donate.sury.org~trusty+2) ...
start: Job failed to start
invoke-rc.d: initscript php7.0-fpm, action "start" failed.
dpkg: error processing package php7.0-fpm (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of php7.0:
 php7.0 depends on php7.0-fpm | libapache2-mod-php7.0 | php7.0-cgi; however:
  Package php7.0-fpm is not configured yet.
  Package libapache2-mod-php7.0 is not installed.
  Package php7.0-cgi is not installed.

dpkg: error processing package php7.0 (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 php7.0-fpm
 php7.0
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by oerdnj** [comment]
@johnluxor What does `php-fpm7.0 -t` says and returns?

**Comment by oerdnj** [comment]
@johnluxor And please don't abuse old closed issues, but open a new issue on a new bug.

**Comment by tomcyr** [comment]
@oerdnj I have same issue:
apt-get upgrade

Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  apache2 apache2-bin apache2-data dbconfig-common dh-php javascript-common
  libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap libhdb9-heimdal
  libjs-codemirror libjs-jquery libjs-jquery-cookie libjs-jquery-event-drag
  libjs-jquery-metadata libjs-jquery-mousewheel libjs-jquery-tablesorter
  libjs-jquery-ui libjs-underscore libkdc2-heimdal libmemcached10 libntdb1
  libpq5 libqdbm14 libx265-59 memcached python-ntdb shtool
Use 'apt-get autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] Y
Setting up php7.0-fpm (7.0.6-3+donate.sury.org~trusty+2) ...
start: Job failed to start
invoke-rc.d: initscript php7.0-fpm, action "start" failed.
dpkg: error processing package php7.0-fpm (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 php7.0-fpm
E: Sub-process /usr/bin/dpkg returned an error code (1)

php-fpm7.0 -t
[10-May-2016 16:37:56.443567] NOTICE: pid 7966, fpm_conf_init_main(), line 1717: configuration file /etc/php/7.0/fpm/php-fpm.conf test is successful

zend_mm_heap corrupted

On Ubuntu 14.04 LTS

**Comment by oerdnj** [comment]
xdebug enabled?

**Comment by tomcyr** [comment]
Yes, xdebug was enabled, I disabled it update is working and php-fpm is starting correctly. Thanks.
